app/lib/QuerySharing.js

```diff
@@ -3,17 +3,18 @@ import markdownTable from 'markdown-table';
 import csvStringify from 'csv-stringify';
 import GitHubApiClient from './GitHubApiClient';
 import Chart from './Chart';
+import DataSource from './DataSource';
 
 export default {
-  async shareOnGist({ query, chart, setting }) {
+  async shareOnGist({ query, chart, setting, dataSource }) {
     let [tsv, svg] = await Promise.all([
       getTableDataAsTsv(query),
       getChartAsSvg(query, chart),
     ]);
 
     let description = query.title;
     let files = {
-      'query.sql': { content: query.body },
+      'query.sql': { content: `/*\ndataSource: ${dataSource.type}\n${DataSource.create(dataSource).description()}\n*/\n\n${query.body}` },
```

Comment:
I think It would be better to separate files like this.
https://gist.github.com/hokaccha/d7562fb49f5004ba5df07bc89ba75d65